### ISSUE TITLE
Broken tests with updated eslint and babel-eslint

### ISSUE THREAD

**Issue by pnevyk** [gajus/eslint-plugin-flowtype#336]
Currently, when `eslint` is updated to `4.19.0` and `babel-eslint` to `8.2.3`, few tests of the plugin are broken. Here is the list:

* **define-flow-type** - Codes containing `interface ...` should throw `no-undef` errors but don't.
* **no-flow-fix-me-comments** - No errors thrown. #307 is related issue.
* **object-type-delimiter** - This seems to be fixed in #317 .
* **sort-keys** - Some variance tests broken. Note that despite that #334 relates to variances in `sort-keys`, the snippet in that issue is fixed incorrectly even with older `eslint` and `eslint-babel`, so it is a separate problem.
* **use-flow-type** - Some `no-unused-vars` tests broken.

These failing tests seem to be blocker for #318.

I will try to look at some of those, but any help is appreciated.

**Comment by gajus** [comment]
> I will try to look at some of those, but any help is appreciated.

I've tried updating those as well but steered away as I've realised it is not a quick fix.

Thank you for investigating.

**Comment by AodhanHayter** [comment]
Good to know others are seeing this as well. I'm getting false positives for `no-unused-vars` concerning imported types, for example:

```javascript
import type Foo from './foo.js' // reported as unused

class Thing {
  constructor(foo: Foo) {
    this.bar = foo.bar
  }
}
```
* `eslint@4.19.1`
* `babel-eslint@8.2.3`
* 'eslint-plugin-flowtype2.49.0`

This problem is remedied in my case by reverting to `babel-eslint@8.0.3`

**Comment by pnevyk** [comment]
So the problem is probably that babel-eslint started to use eslint's new scope manager API ([here](https://github.com/babel/babel-eslint/blob/master/lib/analyze-scope.js)) where a custom Referencer, which visits also type annotations, is implemented. This change was introduced in babel-eslint@8.1.0. This is the reason why our are failing.

This new behavior kind of collides with eslint-plugin-flowtype's `no-undef` and `no-unused-vars` fixes. @gajus Do you have any idea what we should do? Perhaps we can let babel-eslint do the scope tasks eventually.

**Comment by pnevyk** [comment]
I got `no-flow-fix-me-comments` and `sort-keys` working. So the only thing missing is those `no-undef` and `no-unused-vars` errors.

**Comment by gajus** [comment]
> @gajus Do you have any idea what we should do? Perhaps we can let babel-eslint do the scope tasks eventually.

Happy to remove these and release this as a breaking change.

**Comment by pnevyk** [comment]
> Happy to remove these and release this as a breaking change.

OK. I will identify differences between our implementation and babel-eslint because I expect some.

I think you can merge #317 if you have no problems with it. It fixes `object-type-delimiter` rule for new babel-eslint.

**Comment by gajus** [comment]
Closing as stale.
